students/lauraannf/lessons/lesson06/assignment/good_perf.py

```diff
@@ -0,0 +1,59 @@
+"""
+poorly performing, poorly written module
+
+"""
+
+import datetime
+import csv
+import time
+
+
+def analyze(filename):
+    """open and analyze csv file"""
+    start = datetime.datetime.now()
+    start_open1 = time.clock()
+    with open(filename) as csvfile:
+        reader = csv.reader(csvfile, delimiter=',', quotechar='"')
+        found = 0
+        year_count = {
+            "2013": 0,
+            "2014": 0,
+            "2015": 0,
+            "2016": 0,
+            "2017": 0,
+            "2018": 0
+        }
+        for row in reader:
+            if row[5] > '00/00/2012':
+                if row[5][6:] == '2013':
+                    year_count["2013"] += 1
+                if row[5][6:] == '2014':
+                    year_count["2014"] += 1
+                if row[5][6:] == '2015':
+                    year_count["2015"] += 1
+                if row[5][6:] == '2016':
+                    year_count["2016"] += 1
+                if row[5][6:] == '2017':
+                    year_count["2017"] += 1
+                if row[5][6:] == '2018':
+                    year_count["2018"] += 1
+                if "ao" in row[6]:
+                    found += 1
```

Comment:
Also, for cases in which you need to check multiple conditions, you can make use of the _continue_ statement to jump to the next iteration of the for loop, avoiding every single if to be checked after one has already matched:

                if "ao" in row[6]:  # Moved this one up, as it needs to be checked, regardless of whether
                                              # the year matches
                    found += 1
                if row[5][6:] == '2013':
                    year_count["2013"] += 1
                    continue
                if row[5][6:] == '2014':
                    year_count["2014"] += 1
                    continue
                if row[5][6:] == '2015':
                    year_count["2015"] += 1
                    continue
                if row[5][6:] == '2016':
                    year_count["2016"] += 1
                    continue
                if row[5][6:] == '2017':
                    year_count["2017"] += 1
                    continue
                if row[5][6:] == '2018':
                    year_count["2018"] += 1